bpf/lib/nodeport.h

```diff
@@ -1965,10 +2221,34 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 		 */
 		ctx_set_xfer(ctx, XFER_PKT_NO_SVC);
 
+#ifdef ENABLE_DSR
+		if (nodeport_uses_dsr4(&tuple)) {
```

Comment:
I think we need here a comment saying that this is handling a packet to a service which was forwarded by a remote node to us, and that packet has a DSR metadata set.

Reply:
Ah we don't know that yet?

At this point it's merely a packet that **qualifies** for DSR (ie. its L4 protocol has the right type). And we're not just interested in packets that have embedded DSR info, but also non-SYN TCP packets that **match** a DSR connection.

Reply:
> Ah we don't know that yet?

Yep, you are right. Then a comment that a packet **might** have DSR md.

Reply:
Added

> /\* Check if packet has embedded DSR info, or belongs to
>  \* an established DSR connection:
>  \*/